llvm/lib/MC/MCSectionWasm.cpp

```diff
@@ -70,6 +70,8 @@ void MCSectionWasm::printSwitchToSection(const MCAsmInfo &MAI, const Triple &T,
     OS << 'S';
   if (SegmentFlags & wasm::WASM_SEG_FLAG_TLS)
     OS << 'T';
+  if (SegmentFlags & wasm::WASM_SEG_FLAG_NO_STRIP)
+    OS << 'R';
```

Comment:
Does this match ELF?

Reply:
Yes, ELF has a similar flag named `SHF_GNU_RETAIN` and it's represented as `'R'`

Reply:
Should we mirror that name and use `WASM_SEG_FLAG_RETAIN` as the name of our flag?

Reply:
I chose NO_STRIP to align with the existing symbol-level flag WASM_SYMBOL_NO_STRIP, but I don't have strong opinions here.

Reply:
I like to mirror the ELF flag names and I like that that assembly format "R" mnemonic matches, but I don't feel strongly.  @MaskRay WDYT?

Reply:
For the record, it looks like `SHF_MIPS_NOSTRIP` also exists in the ELF world and `SHF_GNU_RETAIN` is a GNU extension rather than part of ELF core.

Reply:
Based on how `WASM_SYMBOL_NO_STRIP ` is chosen as a symbol flag name in https://reviews.llvm.org/D62542#1519799, I agree we don't need to align with it because the segment flag is not based on `.no_dead_strip`. Now I feel `WASM_SEG_FLAG_RETAIN` makes more sense 👍